processing/src/test/java/org/apache/druid/segment/join/HashJoinSegmentStorageAdapterTest.java

```diff
@@ -0,0 +1,1361 @@
+import org.apache.druid.java.util.common.DateTimes;
+import org.apache.druid.java.util.common.Intervals;
+import org.apache.druid.java.util.common.StringUtils;
+import org.apache.druid.java.util.common.granularity.Granularities;
+import org.apache.druid.math.expr.ExprMacroTable;
+import org.apache.druid.query.filter.ExpressionDimFilter;
+import org.apache.druid.query.filter.OrDimFilter;
+import org.apache.druid.query.filter.SelectorDimFilter;
+import org.apache.druid.query.lookup.LookupExtractor;
+import org.apache.druid.segment.QueryableIndexSegment;
+import org.apache.druid.segment.VirtualColumns;
+import org.apache.druid.segment.column.ColumnCapabilities;
+import org.apache.druid.segment.column.ValueType;
+import org.apache.druid.segment.join.lookup.LookupJoinable;
+import org.apache.druid.segment.join.table.IndexedTable;
+import org.apache.druid.segment.join.table.IndexedTableJoinable;
+import org.apache.druid.segment.virtual.ExpressionVirtualColumn;
+import org.apache.druid.timeline.SegmentId;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.IOException;
+import java.util.Collections;
+
+public class HashJoinSegmentStorageAdapterTest
+{
+  private static final String FACT_TO_COUNTRY_ON_ISO_CODE_PREFIX = "c1.";
+  private static final String FACT_TO_COUNTRY_ON_NUMBER_PREFIX = "c2.";
+  private static final String FACT_TO_REGION_PREFIX = "r1.";
+  private static final String REGION_TO_COUNTRY_PREFIX = "rtc.";
+  private static Long NULL_COUNTRY;
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  @Rule
+  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  public QueryableIndexSegment factSegment;
+  public LookupExtractor countryIsoCodeToNameLookup;
+  public LookupExtractor countryNumberToNameLookup;
+  public IndexedTable countriesTable;
+  public IndexedTable regionsTable;
+
+  @BeforeClass
+  public static void setUpStatic()
+  {
+    NullHandling.initializeForTests();
+    NULL_COUNTRY = NullHandling.sqlCompatible() ? null : 0L;
+  }
+
+  @Before
+  public void setUp() throws IOException
+  {
+    factSegment = new QueryableIndexSegment(
+        JoinTestHelper.createFactIndexBuilder(temporaryFolder.newFolder()).buildMMappedIndex(),
+        SegmentId.dummy("facts")
+    );
+    countryIsoCodeToNameLookup = JoinTestHelper.createCountryIsoCodeToNameLookup();
+    countryNumberToNameLookup = JoinTestHelper.createCountryNumberToNameLookup();
+    countriesTable = JoinTestHelper.createCountriesIndexedTable();
+    regionsTable = JoinTestHelper.createRegionsIndexedTable();
+  }
+
+  @After
+  public void tearDown()
+  {
+    if (factSegment != null) {
+      factSegment.close();
+    }
+  }
+
+  @Test
+  public void test_getInterval_factToCountry()
```

Comment:
`factToCountry` isn't really a condition for the test. Is this testing a normal or error case? What kind? I would add also add an expected section to the name like `shouldBeReturned`. Something like `test_getInterval_intervalWithinJoinedSegment_shouldBeReturned()`.